src/network/transaction_request/mod.rs

```diff
@@ -18,75 +19,182 @@ pub struct TransactionRequest {
+    pub fn custom_signature(&self) -> Option<&Bytes> {
+        self.eip_712_meta
+            .as_ref()
+            .and_then(|meta| meta.custom_signature.as_ref())
+    }
+
+    /// Set the custom signature  for the transaction.
     pub fn set_custom_signature(&mut self, custom_signature: Bytes) {
         self.eip_712_meta
             .get_or_insert_with(Eip712Meta::default)
             .custom_signature = Some(custom_signature);
     }
 
+    /// Builder-pattern method for setting custom signature.
     pub fn with_custom_signature(mut self, custom_signature: Bytes) -> Self {
         self.set_custom_signature(custom_signature);
         self
     }
 
+    /// Get the paymaster params for the transaction.
+    pub fn paymaster_params(&self) -> Option<&PaymasterParams> {
+        self.eip_712_meta
+            .as_ref()
+            .and_then(|meta| meta.paymaster_params.as_ref())
+    }
+
+    /// Set the paymaster params  for the transaction.
+    pub fn set_paymaster_params(&mut self, paymaster_params: PaymasterParams) {
+        self.eip_712_meta
+            .get_or_insert_with(Eip712Meta::default)
+            .paymaster_params = Some(paymaster_params);
+    }
+
+    /// Builder-pattern method for setting paymaster params.
+    pub fn with_paymaster_params(mut self, paymaster_params: PaymasterParams) -> Self {
+        self.set_paymaster_params(paymaster_params);
+        self
+    }
+
+    /// Builder-pattern method for building a Zksync EIP-712 create2 tranasaction.
+    pub fn with_create2_params(
+        self,
+        salt: B256,
+        code: Vec<u8>,
+        constructor_data: Vec<u8>,
+        factory_deps: Vec<Vec<u8>>,
+    ) -> Result<Self, BytecodeHashError> {
+        self.with_create_params_inner(Some(salt), code, constructor_data, factory_deps)
+    }
+
+    /// Builder-pattern method for building a ZKsync EIP-712 create tranasaction.
+    pub fn with_create_params(
+        self,
+        code: Vec<u8>,
+        constructor_data: Vec<u8>,
+        factory_deps: Vec<Vec<u8>>,
+    ) -> Result<Self, BytecodeHashError> {
+        self.with_create_params_inner(None, code, constructor_data, factory_deps)
+    }
+
+    pub fn with_create_params_inner(
+        self,
+        salt: Option<B256>,
+        code: Vec<u8>,
+        constructor_data: Vec<u8>,
+        factory_deps: Vec<Vec<u8>>,
+    ) -> Result<Self, BytecodeHashError> {
+        let bytecode_hash = hash_bytecode(&code)?;
+        let factory_deps = factory_deps
+            .into_iter()
+            .chain(vec![code])
+            .map(Into::into)
+            .collect();
+        let input = crate::contracts::l2::contract_deployer::encode_create_calldata(
+            salt,
+            bytecode_hash.into(),
+            constructor_data.into(),
+        );
+        Ok(self
+            .with_to(CONTRACT_DEPLOYER_ADDRESS)
+            .with_input(input)
+            .with_factory_deps(factory_deps))
+    }
+}
+
+impl TransactionRequest {
```

Comment:
Nit: do we need two `impl` blocks?

Reply:
Not really, separated them so it's easier to remove the deprecated methods in the later 0.7.0 update. Can unite them if required. Thoughts?